s2spy/preprocess.py

```diff
@@ -1,11 +1,16 @@
 """Preprocessor for s2spy workflow."""
+import warnings
+from typing import Literal
 from typing import Tuple
 from typing import Union
 import numpy as np
 import scipy.stats
 import xarray as xr
 
 
+TIMESCALE_TYPE = Literal["monthly", "weekly", "daily"]
```

Comment:
I am not so sure about having this alias. Users will see this in the documentation, and when accessing the class/function using `help(...)`:
![image](https://github.com/AI4S2S/s2spy/assets/12114825/ff4f9cfc-34cf-44f2-89e8-27ca8f0b321d)

Perhaps this works better if you define `Timescale` as an alias ([see the python docs](https://docs.python.org/3/library/typing.html#type-aliases)).

```py
from typing import TypeAlias
Timescale: TypeAlias = Literal["monthly", "weekly", "daily"]
```
You can try that, and see if it shows up in the documentation or not.

Reply:
Oh no....`TypeAlias` is available from python 3.10. In python <= 3.9 it should be imported from `typing_extensions`. (https://mypy.readthedocs.io/en/stable/kinds_of_types.html#type-aliases)

Reply:
https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#confval-autodoc_type_aliases it seems like sphinx autodoc does resolve the aliases (unless you explicitly don't want that). 

sphinx-autoapi (what we use) doesn't seem to support this.

Reply:
This is starting to look elegant :) 👀 

Showing the options (Literal['monthly', 'weekly', 'daily']) would be nice in the docs. If I understand correctly, our sphinx-autodoc does not resolve this (yet), and shinx-autoapi does? 

Is it an option to switch? or is there a downside?

Reply:
It is too much to switch from autoapi to autodoc simply for an unresolved alias 😂 , given that they are a bit different and autoapi works quite well for now.

I have added a hint in the tutorial notebook `tutorial_preprocessing.ipynb`:
"User needs to inform preprocessor the temporal resoltuion of their data, by setting the `timescale` argument to be either `daily`, `weekly` or `monthly` timescale."

And we have very clear error message to the user about the supported timescales:

https://github.com/AI4S2S/s2spy/blob/f157ab21ab431ca1cef93d4046abcd09c3f029f4/s2spy/preprocess.py#L131-L138

I think this should be sufficient.